python/plugins/processing/gui/BatchPanel.py

```diff
@@ -217,6 +217,9 @@ def populateByExpression(self, adding=False):
         for k, v in params.items():
             alg_scope.setVariable(k, v, True)
 
+        # add batchCount in the alg scope to be used in the expressions. 0 is only an example value
+        alg_scope.setVariable('row_number', 0, True)
```

Comment:
this should be

    alg_scope.setVariable('row_number', 0, False)

because it's not a static variable -- it's changing with each evaluation

Reply:
right! I missed the real meaning of static... thinking more to an expression evaluation

Reply:
it's used when expressions are prepared -- if a variable is marked as static, it's replaced in the expression internals by a static value to optimise multiple evaluations.